In [40]:
import pandas as pd
import re

# Loading and Exploring dataset

In [41]:
#load dataset of Google Play
df_gg=pd.read_csv("googleplaystore.csv")
df_gg.head(3)

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   

  Reviews  Size    Installs  Type Price Content Rating  \
0     159   19M     10,000+  Free     0       Everyone   
1     967   14M    500,000+  Free     0       Everyone   
2   87510  8.7M  5,000,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up  
2               Art & Design    August 1, 2018       1.2.4  4.0.3 and up

In [42]:
#load dataset of AppStore
df_as=pd.read_csv("AppleStore.csv")
df_as.head(3)

Unnamed: 0         id                                       track_name  \
0           1  281656475                                  PAC-MAN Premium   
1           2  281796108                        Evernote - stay organized   
2           3  281940292  WeatherBug - Local Weather, Radar, Maps, Alerts   

   size_bytes currency  price  rating_count_tot  rating_count_ver  \
0   100788224      USD   3.99             21292                26   
1   158578688      USD   0.00            161065                26   
2   100524032      USD   0.00            188583              2822   

   user_rating  user_rating_ver    ver cont_rating   prime_genre  \
0          4.0              4.5  6.3.5          4+         Games   
1          4.0              3.5  8.2.2          4+  Productivity   
2          3.5              4.5  5.0.0          4+       Weather   

   sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0               38                5        10        1  
1               37                5        23        1  
2               37                5         3        1

In [43]:
#check the data type and null value of the Google play dataset
df_gg.info()
null_gg=df_gg.isnull().sum()
null_gg

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

The dataset of Google Play represents 10841 rows and 13 columns (fields)

There are 1474 null values at Rating field, which is hypothesized that customers are not willing to rate the application after using it. That should be considered and further analyzed.

In [44]:
#check the data type and null value of the App Store dataset
df_as.info()
null_as=df_as.isnull().sum()
null_as

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7197 entries, 0 to 7196
Data columns (total 17 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   Unnamed: 0        7197 non-null   int64  
 1   id                7197 non-null   int64  
 2   track_name        7197 non-null   object 
 3   size_bytes        7197 non-null   int64  
 4   currency          7197 non-null   object 
 5   price             7197 non-null   float64
 6   rating_count_tot  7197 non-null   int64  
 7   rating_count_ver  7197 non-null   int64  
 8   user_rating       7197 non-null   float64
 9   user_rating_ver   7197 non-null   float64
 10  ver               7197 non-null   object 
 11  cont_rating       7197 non-null   object 
 12  prime_genre       7197 non-null   object 
 13  sup_devices.num   7197 non-null   int64  
 14  ipadSc_urls.num   7197 non-null   int64  
 15  lang.num          7197 non-null   int64  
 16  vpp_lic           7197 non-null   int64  


Unnamed: 0          0
id                  0
track_name          0
size_bytes          0
currency            0
price               0
rating_count_tot    0
rating_count_ver    0
user_rating         0
user_rating_ver     0
ver                 0
cont_rating         0
prime_genre         0
sup_devices.num     0
ipadSc_urls.num     0
lang.num            0
vpp_lic             0
dtype: int64

The dataset of App Store encompasses total 7197 rows and 17 columns (fields), which doesn't include any null values

# Data cleaning
## Delete wrong data
The Google Play data set has a dedicated discussion [section](https://www.kaggle.com/datasets/lava18/google-play-store-apps/discussion?sort=hotness), and we can see that one of the discussions outlines an [error](https://www.kaggle.com/datasets/lava18/google-play-store-apps/discussion/164101) for row 10472.

Check the row that has error!

In [45]:
#check data in row 10472
df_gg.iloc[10472].to_frame().T

App Category Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9   19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [46]:
#check unique value of Category field
df_gg["Category"].unique()

array(['ART_AND_DESIGN', 'AUTO_AND_VEHICLES', 'BEAUTY',
       'BOOKS_AND_REFERENCE', 'BUSINESS', 'COMICS', 'COMMUNICATION',
       'DATING', 'EDUCATION', 'ENTERTAINMENT', 'EVENTS', 'FINANCE',
       'FOOD_AND_DRINK', 'HEALTH_AND_FITNESS', 'HOUSE_AND_HOME',
       'LIBRARIES_AND_DEMO', 'LIFESTYLE', 'GAME', 'FAMILY', 'MEDICAL',
       'SOCIAL', 'SHOPPING', 'PHOTOGRAPHY', 'SPORTS', 'TRAVEL_AND_LOCAL',
       'TOOLS', 'PERSONALIZATION', 'PRODUCTIVITY', 'PARENTING', 'WEATHER',
       'VIDEO_PLAYERS', 'NEWS_AND_MAGAZINES', 'MAPS_AND_NAVIGATION',
       '1.9'], dtype=object)

The problem of data in that row is the mismatching between header and value in Category and Rating fields. Because the maxinimun rating score in Google Play is 5, but data is 19. And, data in Category field is "1.9" while it should be one of the above category such as "art and design", "auto and vehicles", "Beaty", and so on.

The solution for that row is that it should be deleted not to affect the results of analysis.

In [47]:
#delete row 10472
df_gg=df_gg.drop(10472)

#check to know whether the row 10472 was deleted
total_rows,total_cols= df_gg.shape
print("Total rows:", total_rows)

Total rows: 10840


## Check and delete duplicate values

### Check and delete duplicate values in Google Play dataset

In [48]:
#check the total number of duplicate values
duplicate=df_gg.duplicated(subset= ["App","Category"], keep=False)
total_dup= duplicate.sum()
total_dup

1847

Duplicate apps are defined as apps that have the same app name and category. There are 1847 duplicate values are found in Google Play dataset.

To make sure that the dataset show the meaningful results for analysis, delete duplicate apps that have lower rating score and keep the apps with higher rating score

In [49]:
#sort dataset by App, Category, and Rating
df_gg=df_gg.sort_values(by=["App", "Category","Rating"], ascending=[True, True, False])

#delete duplicate values
df_gg= df_gg.drop_duplicates(subset=["App", "Category"], keep="first")
df_gg.shape[0]

9744

The Google Play dataset remains 9744 rows after deleting duplicate values

### Check and delete duplicate values in App store dataset

In [50]:
# check the total number of duplicate values
duplicate= df_as.duplicated(subset=["track_name", "prime_genre"], keep=False)
total_dup=duplicate.sum()
total_dup

4

In [51]:
# delete duplicate values
df_as= df_as.drop_duplicates(subset=["track_name", "prime_genre"], keep="first")
print("Total rows of App Store data:", df_as.shape[0])

Total rows of App Store data: 7195


## Check and remove Non-English Apps

Because the Non-English apps which is defined as the ones that have non-English name aim at the different target audiences from whom we target to offer. We will check whether the dataset includes non-English apps. If there are some non-English apps, remove them

### Check and remove Non-English apps on Google Play dataset

In [52]:
# Use regular expressions library to check whether app name is English

# Define function
def check_non_english_app(app_name):
    # Regex to match characters outside a-z, A-Z, 0-9, and basic punctuation
    return len(re.findall(r'[^\x00-\x7F]', app_name))

# Apply the check_non_english_app function
non_english_app=df_gg[df_gg["App"].apply(check_non_english_app) >=3]
print("Total Non-English apps:", non_english_app.shape[0])

Total Non-English apps: 62


Using the Regular expressions library to define a function that runs through each line of data to check if the app name is in English or not. To minimize the impact of data cleaning on results of analysis, I consider apps that have more than or equal to 3 non-ASCII characters as non-English apps which need removing from the Google Play dataset 

The result shows that there are 62 Non_English apps which need removing from the Google Play dataset

In [53]:
# Remove Non-English apps
df_gg=df_gg[df_gg["App"].apply(check_non_english_app)<3]
print("Total Non-English apps:", df_gg.shape[0])

Total Non-English apps: 9682


After removing non-English apps, wrong data, and duplicate values as well, Google Play dataset has 9682 rows that are ready to further analysis.

### Check and remove Non-English apps on App Store dataset

In [54]:
# Check non-English apps in App Store dataset using check_non_english_app function defined above

non_english_app=df_as[df_as["track_name"].apply(check_non_english_app) >=3]
print("Total number of Non-English apps on App Store:", non_english_app.shape[0])

Total number of Non-English apps on App Store: 1042


In [55]:
# Remove non-English apps from App Store dataset
df_as=df_as[df_as["track_name"].apply(check_non_english_app) <3]
df_as.shape[0]

6153

After removing non-English apps, wrong data, and duplicate values as well, App store dataset has 6153 rows that are ready to further analysis.

## Isolate Free Apps
As mentioned in the Overview and Objective sections, the project that analysis supports is an ad-supported mobile app - a free app profiting from in-app ads. So, only data about free apps should be took into account

Our data sets contain both free and non-free apps, and we'll need to isolate only the free apps for our analysis. Below, we isolate the free apps for both our data sets.

In [56]:
# Create a dataframe of free apps data
free_apps_gg=df_gg[df_gg["Price"]=="0"]
free_apps_as=df_as[df_as["price"]==0]

print("Total Google Play free apps:", free_apps_gg.shape[0])
print("Total App Store free apps:", free_apps_as.shape[0])

Total Google Play free apps: 8929
Total App Store free apps: 3201


At the end of Data cleaning section, there are 8929 rows on Google Play and 3201 rows on App Store are ready for further analysis

# Most common Free apps by Genre
The objectives of analysis is to determine the types of apps that are likely to **attract tremendous of users** because, as discussed above, revenue is highly influenced by the number of people using apps.

To minimize risks and optimize ROI, the product development plan comprises 3 phases:
- Build a MVP app in Android version, then add it to Google Play
- Observe and evaluate responses of users to improve UX-UI and other system features of the app
- Build an iOS version of app if the app is profitable (or ROI > 1) after 6 months

Followed by this plan, the app will be added to both the Google Play and App Store, so we need to find app profiles that are successful on both markets. The features that we will focus on when analyze app profiles are **Genre** or **Category** of the apps

Since the primary goal is to determine which app genres are **the most popular** or **downloaded the most**, analyzing the *frequency* of genres and *the number of downloads* by Genre is a suitable solution. The fields of Rating and Comment are not necessary for in-depth analysis now.

## Explore frequency by Genre and Category on Google Play and App Store free apps

In [57]:
# Create frequency table by Genre of Google Play free apps
frequency_table_gg= free_apps_gg.groupby(["Genres"]).size().reset_index(name="Frequency")

# Calculate the % of frequency
frequency_table_gg["Percentage"]= (frequency_table_gg["Frequency"]/len(free_apps_gg))*100
frequency_table_gg= frequency_table_gg.sort_values(by="Percentage", ascending=False)
frequency_table_gg

Genres  Frequency  Percentage
104                        Tools        747    8.365998
49                 Entertainment        556    6.226901
36                     Education        483    5.409340
19                      Business        407    4.558181
79                  Productivity        345    3.863815
..                           ...        ...         ...
84              Puzzle;Education          1    0.011199
66        Lifestyle;Pretend Play          1    0.011199
70   Music & Audio;Music & Video          1    0.011199
4            Adventure;Education          1    0.011199
32             Comics;Creativity          1    0.011199

[115 rows x 3 columns]

The top 5 genres with the largest number of free apps on Google Play are Tools, Entertainment, Education, Business, and Productivity, with frequencies that are not significantly different from each other. This shows that app genres supporting users in work and life are as diverse as entertainment apps

In [58]:
# Create frequency table by Category of Google Play free apps
frequency_cate_gg= free_apps_gg.groupby(["Category"]).size().reset_index(name="Frequency")

# Calculate the % of frequency
frequency_cate_gg["Percentage"] = (frequency_cate_gg["Frequency"] / len(free_apps_gg))*100
frequency_cate_gg= frequency_cate_gg.sort_values(by="Percentage", ascending= False)
frequency_cate_gg.head(10)

Category  Frequency  Percentage
11           FAMILY       1710   19.151081
14             GAME        872    9.765931
29            TOOLS        748    8.377198
4          BUSINESS        407    4.558181
25     PRODUCTIVITY        345    3.863815
18        LIFESTYLE        344    3.852615
12          FINANCE        328    3.673424
20          MEDICAL        313    3.505432
28           SPORTS        302    3.382238
23  PERSONALIZATION        294    3.292642

The difference between the **Genres** and the **Category** columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). The overview picture of category shows that the majority of free apps on Google Play relate to Family topics

In [59]:
# Create frequency table by Genre of App Store free apps
frequency_table_as= free_apps_as.groupby(by="prime_genre").size().reset_index(name="Frequency")

# Calculate percentage
frequency_table_as["Percentage"] = (frequency_table_as["Frequency"] / len(free_apps_as))*100
frequency_table_as=frequency_table_as.sort_values(by="Percentage", ascending=False)
frequency_table_as.head(10)

prime_genre  Frequency  Percentage
7               Games       1864   58.231803
4       Entertainment        251    7.841300
14      Photo & Video        160    4.998438
3           Education        118    3.686348
18  Social Networking        106    3.311465
17           Shopping         83    2.592940
21          Utilities         79    2.467979
19             Sports         69    2.155576
11              Music         66    2.061856
8    Health & Fitness         65    2.030615

Most free apps on the App Store are Games, accounting for 58% of all free apps. This reveals a significant imbalance in the distribution of free app genres on the App Store, where for-fun apps is the dominant genre.

## Evaluate users interest through downloads by Genre on Google Play

Use **Installs** field to analyze customers interest by Genre

In [60]:
# Check data type of Installs field
free_apps_gg["Installs"].head(5)

8884          500+
8532    1,000,000+
324        10,000+
4541       10,000+
4636       10,000+
Name: Installs, dtype: object

In [61]:
# Remove the "+" at the end of values
free_apps_gg["Installs"]=free_apps_gg["Installs"].str.rstrip("+")

# Remove the "," character
free_apps_gg["Installs"]=free_apps_gg["Installs"].str.replace(",","", regex=False)

# Convert Installs field into float
free_apps_gg["Installs"]=free_apps_gg["Installs"].astype(float)

C:\Users\hangu\AppData\Local\Temp\ipykernel_14556\2220757888.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  free_apps_gg["Installs"]=free_apps_gg["Installs"].str.rstrip("+")
C:\Users\hangu\AppData\Local\Temp\ipykernel_14556\2220757888.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  free_apps_gg["Installs"]=free_apps_gg["Installs"].str.replace(",","", regex=False)
C:\Users\hangu\AppData\Local\Temp\ipykernel_14556\2220757888.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a sl

In [62]:
# Calculate the average number of downloads by Genre
avg_downloads=free_apps_gg.groupby(by="Genres")["Installs"].mean().reset_index(name="Avg_num_downloads").sort_values(by="Avg_num_downloads", ascending=False)
avg_downloads.head(30)

Genres  Avg_num_downloads
3      Adventure;Action & Adventure       5.150000e+07
33                    Communication       3.845960e+07
24                           Casual       2.995555e+07
110         Video Players & Editors       2.494734e+07
97                           Social       2.325365e+07
5                            Arcade       2.302817e+07
25        Casual;Action & Adventure       1.961538e+07
81        Puzzle;Action & Adventure       1.836667e+07
78                      Photography       1.780563e+07
44   Educational;Action & Adventure       1.701667e+07
79                     Productivity       1.678733e+07
85                           Racing       1.591065e+07
100                        Strategy       1.473479e+07
106                  Travel & Local       1.405148e+07
0                            Action       1.246711e+07
104                           Tools       1.083136e+07
80                           Puzzle       1.048822e+07
105                 Tools;Education       1.000000e+07
66           Lifestyle;Pretend Play       1.000000e+07
90         Role Playing;Brain Games       1.000000e+07
21          Card;Action & Adventure       1.000000e+07
29             Casual;Music & Video       1.000000e+07
4               Adventure;Education       1.000000e+07
72                 News & Magazines       9.549178e+06
69                            Music       9.445583e+06
48         Educational;Pretend Play       9.375000e+06
82               Puzzle;Brain Games       9.013125e+06
30              Casual;Pretend Play       8.913636e+06
86        Racing;Action & Adventure       8.816667e+06
17                Books & Reference       8.814200e+06

The top 5 genres with the highest download numbers on Google Play are Adventure, Communication, Casual, Video Players & Editors, and Social. However, it's essential to examine whether a few outlier apps are dominating the download numbers, potentially distorting the overall picture.

In [63]:
# Calculate the average number of downloads by Genre in FAMILY category
family= free_apps_gg[free_apps_gg["Category"]=="FAMILY"]
family_num= family.groupby(by="Genres")["Installs"].mean().reset_index(name="Total_downloads").sort_values(by="Total_downloads", ascending=False)
family_num.head(10)

Genres  Total_downloads
1     Adventure;Action & Adventure     3.533333e+07
13                          Casual     1.952373e+07
47       Puzzle;Action & Adventure     1.836667e+07
29  Educational;Action & Adventure     1.701667e+07
14       Casual;Action & Adventure     1.291667e+07
62                        Strategy     1.157132e+07
2              Adventure;Education     1.000000e+07
55        Role Playing;Brain Games     1.000000e+07
18            Casual;Music & Video     1.000000e+07
23           Education;Brain Games     1.000000e+07

Casual is a standout genre, with average downloads consistently ranking in the Top 3 across both the overall genre popularity and the FAMILY category which boasts the largest number of free apps. This suggests that Casual games have a strong appeal to users, making it a genre worth further exploration.

In [64]:
# Whether a few dominate apps skew the average number of Adventure genre
Adventure= free_apps_gg[free_apps_gg["Genres"]=="Adventure;Action & Adventure"]
Adventure_apps= Adventure.groupby(by="App")["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending=False)
Adventure_apps

App   Total_down
1                                     ROBLOX  200000000.0
2  Transformers Rescue Bots: Hero Adventures    5000000.0
0                                Leo and Tig    1000000.0

There are just 3 free apps belonging to Adventure genre and the number of downloads of ROBLOX app is dominent. The findings indicate that the Adventure genre lacks diversity in its data, making it challenging to draw meaningful insights. Any conclusions drawn from this limited data would likely be biased.

Therefore, it is recommended to revisit the analysis of this genre once the dataset has been expanded. For now, the Adventure genre will be excluded from further consideration

In [65]:
# Whether a few dominate apps skew the average number of Communication genre
communication= free_apps_gg[free_apps_gg["Genres"]=="Communication"]
communication_apps= communication.groupby(by="App")["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending=False)
communication_apps.head(10)

App    Total_down
148              Google Chrome: Fast & Secure  1.000000e+09
185  Messenger – Text and Video Chat for Free  1.000000e+09
154                                  Hangouts  1.000000e+09
145                                     Gmail  1.000000e+09
225             Skype - free IM & video calls  1.000000e+09
258                        WhatsApp Messenger  1.000000e+09
149     Google Duo - High Quality Video Calls  5.000000e+08
246                           Viber Messenger  5.000000e+08
278             imo free video calls and chat  5.000000e+08
172               LINE: Free Calls & Messages  5.000000e+08

The Communication genre is dominated by large global companies, with no other notable free apps in this space.

Further analysis of this genre will not be pursued, as the dominant companies in this industry are not competitors that can be easily surpassed, and it would be challenging to create a competitive advantage in this niche market.

In [66]:
# Explore the Casual genre on Google Play
casual= free_apps_gg[free_apps_gg["Genres"]=="Casual"]
casual_apps= casual.groupby(by="App")["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending=False)
casual_apps.head(10)

App    Total_down
43        Candy Crush Saga  1.000000e+09
124                    Pou  1.000000e+09
108         My Talking Tom  1.000000e+09
71        Farm Heroes Saga  2.000000e+08
107      My Talking Angela  2.000000e+08
13           Angry Birds 2  2.000000e+08
44   Candy Crush Soda Saga  1.000000e+08
152                Yes day  1.000000e+08
141        Talking Tom Cat  1.000000e+08
35     Bubble Witch 2 Saga  1.000000e+08

Casual is a genre with the impressive number of downloads, mainly consisting of time-killing games and games for all ages. This reflects users' demand for short-term entertainment and mid-break relaxation, presenting an opportunity for businesses to further exploit. Moreover, the probability of users clicking on in-app ads while playing time-killing games is higher compared to using other apps. Click-through rate of gaming segment is 11.4%, which is the highest among all categories (source: [embryo](https://embryo.com/blog/in-app-stats/))

The top-downloaded games of that genre are puzzle games and voice-interactive games. Meanwhile, the niche of time-killing and all ages games is diverse. There are many game themes that companies can explore.

Continue analyzing genres such as Video Players & Editors, Social, and others to gain insights into new development directions for casual games and games for all ages.

In [67]:
# Explore the Video Players & Editors genre on Google Play
vid_players= free_apps_gg[free_apps_gg["Genres"]=="Video Players & Editors"]
vid_players_apps= vid_players.groupby(["App"])["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending=False)
vid_players_apps.head(10)

App    Total_down
76                             Google Play Movies & TV  1.000000e+09
145                                            YouTube  1.000000e+09
93                                           MX Player  5.000000e+08
137             VivaVideo - Video Editor & Photo Movie  1.000000e+08
134  VideoShow-Video Editor, Video Maker, Beauty Ca...  1.000000e+08
120                                    VLC for Android  1.000000e+08
100                                   Motorola Gallery  1.000000e+08
99                                   Motorola FM Radio  1.000000e+08
55                                            Dubsmash  1.000000e+08
96         MiniMovie - Free Video and Slideshow Editor  5.000000e+07

In [68]:
# Explore the Social genre on Google Play
social=free_apps_gg[free_apps_gg["Genres"]=="Social"]
social_apps= social.groupby(["App"])["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending=False)
social_apps.head(10)

App    Total_down
111                       Instagram  1.000000e+09
100                         Google+  1.000000e+09
75                         Facebook  1.000000e+09
176                        Snapchat  5.000000e+08
77                    Facebook Lite  5.000000e+08
16          BIGO LIVE - Live Stream  1.000000e+08
123                        LinkedIn  1.000000e+08
199                          Tumblr  1.000000e+08
197  Tik Tok - including musical.ly  1.000000e+08
160                       Pinterest  1.000000e+08

Video Players & Editors and Social genres have a similar pattern to Communication, with large companies dominating the market and limited opportunities for new insights.

In [69]:
# Explore Travel & Local genre on Google Play
travel=free_apps_gg[free_apps_gg["Genres"]=="Travel & Local"]
travel_apps= travel.groupby(["App"])["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending=False)
travel_apps.head(10)

App    Total_down
107                                 Google Street View  1.000000e+09
136                          Maps - Navigate & Explore  1.000000e+09
173  TripAdvisor Hotels Flights Restaurants Attract...  1.000000e+08
106                                       Google Earth  1.000000e+08
35                            Booking.com Travel Deals  1.000000e+08
205                           trivago: Hotels & Travel  5.000000e+07
1                          2GIS: directory & navigator  5.000000e+07
133        MAPS.ME – Offline Map and Travel Navigation  5.000000e+07
182                                       VZ Navigator  5.000000e+07
162                                         Skyscanner  1.000000e+07

In [70]:
# Explore Lifestyle genre on Google Play
lifestyle=free_apps_gg[free_apps_gg["Genres"]=="Lifestyle;Pretend Play"]
lifestyle_apps= lifestyle.groupby(["App"])["Installs"].sum().reset_index(name="Total_down").sort_values(by="Total_down", ascending= False)
lifestyle_apps.head(10)

App  Total_down
0  Talking Babsy Baby: Baby Games  10000000.0

In [71]:
# Explore Books & Reference genre on Google Play
book=free_apps_gg[free_apps_gg["Genres"]=="Books & Reference"]
book_apps= book.groupby(["App"])["Installs"].sum().reset_index(name="total_down").sort_values(by="total_down", ascending=False)
book_apps

App    total_down
119        Google Play Books  1.000000e+09
42                     Bible  1.000000e+08
30   Audiobooks from Audible  1.000000e+08
181     Wattpad 📖 Free Books  1.000000e+08
25             Amazon Kindle  1.000000e+08
..                       ...           ...
141      Learn R Programming  1.000000e+01
60                   CZ-Help  5.000000e+00
125  Guide for R Programming  5.000000e+00
46    BibleRead En Cy Zh Yue  5.000000e+00
90                    EW PDF  5.000000e+00

[189 rows x 2 columns]

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average of the genre. However, it looks like there are only a few very popular apps (Google Play Books, Bible, Wattpad free books), so this market still shows potential.

With the potential interest of users in Book & Reference genre and Casual genre, there is a promising return on investment (ROI) if an ad-supported mobile app is developed by **innovatively integrating the characteristics of a reading app and a casual game, catering to users of all ages**. It maybe a mobile game that challenge users on puzzle arround books content with other features to help users practice language or programming.

Let's move on to explore users interest by genre on App Store to determine which is the most optimal recommendation for developing ad-supported mobile app.

## Evaluate users interest through downloads by Genre on App Store
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

In [72]:
# Calculate the average number of rating by Genre on App Store
avg_rating=free_apps_as.groupby(["prime_genre"])["rating_count_tot"].mean().reset_index(name="Avg_num_rating").sort_values(by="Avg_num_rating", ascending=False)
avg_rating.head(10)

prime_genre  Avg_num_rating
12         Navigation    86090.333333
16          Reference    79350.470588
18  Social Networking    71548.349057
11              Music    57326.530303
22            Weather    52279.892857
0                Book    46384.916667
6        Food & Drink    33333.923077
5             Finance    32367.028571
14      Photo & Video    28441.543750
20             Travel    28243.800000

In [73]:
# Calculate the average number of rating by App of each Genre
# rating by App on Navigation genre
navigate= free_apps_as[free_apps_as["prime_genre"]=="Navigation"]
navigate_rating = navigate.groupby(["track_name"])["rating_count_tot"].mean().reset_index(name="Avg_num_rating").sort_values(by="Avg_num_rating", ascending=False)

# rating by App on Reference genre
reference= free_apps_as[free_apps_as["prime_genre"]=="Reference"]
reference_rating = reference.groupby(["track_name"])["rating_count_tot"].mean().reset_index(name="Avg_num_rating").sort_values(by="Avg_num_rating", ascending=False)

# rating by App on Socical Networking
social= free_apps_as[free_apps_as["prime_genre"]=="Social Networking"]
social_rating = social.groupby(["track_name"])["rating_count_tot"].mean().reset_index(name="Avg_num_rating").sort_values(by="Avg_num_rating", ascending=False)

# rating by App on Music
music= free_apps_as[free_apps_as["prime_genre"]=="Music"]
music_rating = music.groupby(["track_name"])["rating_count_tot"].mean().reset_index(name="Avg_num_rating").sort_values(by="Avg_num_rating", ascending=False)

# rating by App on Book
book= free_apps_as[free_apps_as["prime_genre"]=="Book"]
book_rating = book.groupby(["track_name"])["rating_count_tot"].mean().reset_index(name="Avg_num_rating").sort_values(by="Avg_num_rating", ascending=False)


In [74]:
# Show rating by App on Navigation genre
navigate_rating.head(10)

track_name  Avg_num_rating
5   Waze - GPS Navigation, Maps & Real-time Traffic        345046.0
2                Google Maps - Navigation & Transit        154911.0
1                                       Geocaching®         12811.0
0       CoPilot GPS – Car Navigation & Offline Maps          3582.0
3  ImmobilienScout24: Real Estate Search in Germany           187.0
4                              Railway Route Search             5.0

In [75]:
# Show rating by App on Social
social_rating.head(10)

track_name  Avg_num_rating
16                                  Facebook       2974676.0
63                                 Pinterest       1061624.0
78                          Skype for iPhone        373519.0
55                                 Messenger        351466.0
85                                    Tumblr        334293.0
91                        WhatsApp Messenger        287589.0
41                                       Kik        260965.0
103  ooVoo – Free Video Call, Text and Voice        177501.0
83          TextNow - Unlimited Text + Calls        164963.0
86             Viber Messenger – Text & Call        164249.0

In [76]:
# Show rating by App on Music
music_rating.head(10)

track_name  Avg_num_rating
37                            Pandora - Music & Radio       1126879.0
56                                      Spotify Music        878563.0
45  Shazam - Discover music, artists, videos & lyrics        402925.0
64          iHeartRadio – Free Music & Radio Stations        293228.0
52                         SoundCloud - Music & Audio        135744.0
24                               Magic Piano by Smule        131695.0
49                                        Smule Sing!        119316.0
60   TuneIn Radio - MLB NBA Audiobooks Podcasts Music        110420.0
2                                        Amazon Music        106235.0
53              SoundHound Song Search & Music Player         82602.0

On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews.

The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

In [77]:
# Show rating by App on Book
book_rating.head(10)

track_name  Avg_num_rating
6         Kindle – Read eBooks, Magazines & Textbooks        252076.0
0   Audible – audio books, original series & podcasts        105274.0
2        Color Therapy Adult Coloring Book for Adults         84062.0
9           OverDrive – Library eBooks and Audiobooks         65450.0
5                               HOOKED - Chat Stories         47829.0
1   BookShout: Read eBooks & Track Your Reading Goals           879.0
3             Dr. Seuss Treasury — 50 best kids books           451.0
4                                   Green Riding Hood           392.0
10                                    Weirdwood Manor           197.0
8                            MangaZERO - comic reader             9.0

In [78]:
# Show rating by App on Reference genre
reference_rating.head(10)

track_name  Avg_num_rating
0                                               Bible        985920.0
2               Dictionary.com Dictionary & Thesaurus        200047.0
3      Dictionary.com Dictionary & Thesaurus for iPad         54175.0
5                                    Google Translate         26786.0
11  Muslim Pro: Ramadan 2017 Prayer Times, Azan, Q...         18418.0
12  New Furniture Mods - Pocket Wiki & Game Tools ...         17588.0
10                         Merriam-Webster Dictionary         16849.0
13                                          Night Sky         12122.0
1   City Maps for Minecraft PE - The Best Maps for...          8535.0
9   LUCKY BLOCK MOD ™ for Minecraft PC Edition - T...          4693.0

However, the niche of Book and Reference seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

- Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

- Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

- Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

# Conclusion
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can attract more and more users to download on both platforms.

With the potential interest of users in Book & Reference genre and Casual genre in both Google Play and App Store, developing an ad-supported mobile app that combines the features of a reading app and a casual game *for all ages*, focusing on puzzle challenges related to book content, language practice, programming, or finance could yield a promising profit for our company.